src/ceph-volume/plugin/zfs/ceph_volume_zfs/util/prepare.py

```diff
@@ -0,0 +1,260 @@
+"""
+These utilities for prepare provide all the pieces needed to prepare a device
+but also a compounded ("single call") helper to do them in order. Some plugins
+may want to change some part of the process, while others might want to consume
+the single-call helper
+"""
+import os
+import logging
+import json
+from ceph_volume import process, conf
+from ceph_volume.util import system, constants
+
+logger = logging.getLogger(__name__)
```

Comment:
@tchaikov @alfredodeza 
I agree with the comment, but IMHO that then also would require pushing the parts that are in the `global` prepare code down into the `LVM` code. Since they are also not required globaly.

So that is why I have a ZFS version of util prepare, and have kept the few duplicated commands.
But it is possible to take them out and use the `global` variants.

Reply:
@wjwithagen I think we can do better here by generalizing stuff that things like this plugin could make use of. While we identify those, lets try to re-use as much as possible